utils/python/CIME/hist_utils.py

```diff
@@ -242,19 +247,29 @@ def get_extension(model, filepath):
     'hi'
     >>> get_extension("cpl", "TESTRUNDIFF_Mmpi-serial.f19_g16_rx1.A.melvin_gnu.C.fake_testing_only_20160816_164150-20160816_164240.cpl.h.nc")
     'h'
+    >>> get_extension("clm","clm2_0002.h0.1850-01-06-00000.nc")
+    '0002.h0'
     """
     basename = os.path.basename(filepath)
-    ext_regex = re.compile(r'.*%s.*[.](h.?)([.][^.]+)?[.]nc' % model)
+    ext_regex = re.compile(r'.*%s[^_]*_?(\d{4})?[.](h.?)([.][^.]+)?[.]nc' % model)
```

Comment:
Here you use \d{4} for the multi-instance suffix, whereas in two other places you use [0-9]{4}. I think those mean the same thing, but it would be helpful for future understanding and maintenance if you were consistent in all three places. (Even better if you could somehow store that in a shared variable, but I'm not sure how easy that is, so I won't push on that.)